test/unit/spec/util/jobLogViewerSpec.js

```diff
@@ -48,217 +222,493 @@ define(['util/jobLogViewer', 'common/runtime'], (JobLogViewer, Runtime) => {
-                                {
-                                    is_error: 0,
-                                    line: 'line 1 - log',
-                                    linepos: 1,
-                                    ts: 123456789,
-                                },
-                                {
-                                    is_error: 1,
-                                    line: 'line 2 - error',
-                                    linepos: 1,
-                                    ts: 123456790,
-                                },
-                            ],
-                        },
-                    },
-                    {
-                        channel: {
-                            jobId: jobId,
-                        },
-                        key: {
-                            type: 'job-logs',
-                        },
+        describe('response to update', () => {
+            [true, false].forEach((mode) => {
+                beforeEach(function () {
+                    if (mode) {
+                        this.jobLogViewerInstance = JobLogViewer.make({
+                            showHistory: mode,
+                            devMode: true,
+                        });
                     }
-                );
-                setTimeout(() => {
-                    const panel = hostNode.querySelector('[data-element="log-panel"]');
-                    expect(panel.children.length).toEqual(2);
-                    const logLine = panel.children[0];
-                    expect(logLine.classList.toLocaleString()).toContain(
-                        `${cssBaseClass}__line_container`
-                    );
-                    expect(logLine.innerHTML).toContain('line 1 - log');
-                    const errorLine = panel.children[1];
-                    expect(errorLine.classList.toLocaleString()).toContain(
-                        `${cssBaseClass}__line_container--error`
-                    );
-                    expect(errorLine.innerHTML).toContain('line 2 - error');
-                    viewer.detach();
-                    done();
-                }, 500);
+                });
+
+                JobsData.validJobs.forEach((jobState) => {
+                    it(`should create a string for status ${jobState.status}, history mode ${
+                        mode ? 'on' : 'off'
+                    }`, async function () {
+                        this.runtimeBus.on('request-job-status', (msg) => {
+                            testJobStatus(this, mode);
+                            expect(msg).toEqual({ jobId: jobState.job_id });
+                        });
+
+                        await this.jobLogViewerInstance.start({
+                            node: this.node,
+                            jobId: jobState.job_id,
+                        });
+
+                        return createStatusObserver(this, formatStatusMessage(jobState)).then(
```

Comment:
This promise resolves when there is a change under the job status node -- i.e. the new state is displayed in the UI.